libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2515,8 +2515,47 @@ void TypeChecker::requireLValue(Expression const& _expression)
 	_expression.annotation().lValueRequested = true;
 	_expression.accept(*this);
 
-	if (_expression.annotation().isConstant)
-		m_errorReporter.typeError(_expression.location(), "Cannot assign to a constant variable.");
-	else if (!_expression.annotation().isLValue)
-		m_errorReporter.typeError(_expression.location(), "Expression has to be an lvalue.");
+	if (_expression.annotation().isLValue)
+		return;
+
+	return m_errorReporter.typeError(_expression.location(), [&]() {
```

Comment:
What about this :-)? Now I don't need to scroll down to know that nothing else is happening and it's valid to return void in a situation like this :-).

Reply:
Ugh, but ok...